integration/integration_test.go

```diff
@@ -457,16 +458,32 @@ func TestDns(t *testing.T) {
 	t.Parallel()
 
 	dnsCmd := &command{
-		name:           "StartDnsGadget",
-		cmd:            fmt.Sprintf("$KUBECTL_GADGET trace dns -n %s", ns),
-		expectedRegexp: fmt.Sprintf(`%s\s+test-pod\s+OUTGOING\s+A\s+microsoft.com`, ns),
-		startAndStop:   true,
+		name:         "StartDnsGadget",
+		cmd:          fmt.Sprintf("$KUBECTL_GADGET trace dns -n %s -o json", ns),
+		startAndStop: true,
+		expectedOutputFn: func(output string) error {
+			expectedEntry := &dnsTypes.Event{
+				Event:   buildBaseEvent(ns),
+				PktType: "OUTGOING",
+				DNSName: "inspektor-gadget.io.",
+				QType:   "A",
+			}
+
+			// DNS gadget doesn't provide container data. Remove it.
+			expectedEntry.Container = ""
+
+			normalize := func(e *dnsTypes.Event) {
+				e.Node = ""
+			}
+
+			return expectAllToMatch(output, normalize, expectedEntry)
+		},
 	}
 
 	commands := []*command{
 		createTestNamespaceCommand(ns),
 		dnsCmd,
-		busyboxPodRepeatCommand(ns, "nslookup microsoft.com"),
+		busyboxPodRepeatCommand(ns, "nslookup -query=A inspektor-gadget.io"),
```

Comment:
Why do you prefer generating only IPv4 DNS requests? We could use expectEntriesToMatch to verify `A` and `AAAA` requests are generated.

Reply:
Right. I didn't think about `expectEntriesToMatch`.